packages/@vue/cli/lib/options.js

```diff
@@ -10,8 +10,6 @@ const rcPath = exports.rcPath = getRcPath('.vuerc')
 const presetSchema = createSchema(joi => joi.object().keys({
   bare: joi.boolean(),
   useConfigFiles: joi.boolean(),
-  router: joi.boolean(),
```

Comment:
Though not preferred, they should remain in the schema, otherwise the legacy preset format cannot pass the schema validation.

Note: you may leave a TODO: comment here, once `@hapi/joi` v16 released, we can update the schema to utilizing its `.warn` feature. https://github.com/hapijs/joi/commit/59e60aa851f674d63c9ac1bad4b0f59c0fa24033